core/src/stage.cpp

```diff
@@ -283,6 +283,13 @@ void Stage::setName(const std::string& name) {
 	pimpl_->name_ = name;
 }
 
+uint32_t Stage::introspectionId() const {
+	if (!pimpl_->introspection_)
+		throw std::runtime_error(
+		    "Introspection not available. Initialize the full Task before calling introspectionId()");
+	return const_cast<const moveit::task_constructor::Introspection*>(pimpl_->introspection_)->stageId(this);
```

Comment:
Why is it necessary to explicitly cast to `const` here? As the method is `const`, all members of this should be const as well, shouldn't they? Or do you want to make it explicit?
```suggestion
	return pimpl_->introspection_->stageId(this);
```

Reply:
It actually fails to compile here without the const cast because calling `stageId` on a non-const introspection requires private visibility (in contrast to the const version) and gcc's implementation rejects the call because of that.

The private method would actually *name* the stage (give it a new number) without checking whether it already has one.

Reply:
Yes, I was assuming that the introspection instance should be const because of the main method being const...

Reply:
No, it's just the `introspection_` pointer that's const.